contracts/farm/LiquidityMiningV2.sol

```diff
@@ -925,4 +925,79 @@ contract LiquidityMiningV2 is ILiquidityMining, LiquidityMiningStorageV2 {
 		uint256 poolId = _getPoolId(_poolToken);
 		return userInfoMap[poolId][_user].amount;
 	}
+
+	/**
+	 * @notice read all pools from liquidity mining V1 contract and add them
+	 * @param _lmContract the address of the liquidity mining V1 contract
+	 */
+	function migratePools(address _lmContract, address _SOVAddress) external onlyAuthorized {
+		require(_lmContract != address(0), "Invalid contract address");
+		require(_SOVAddress != address(0), "Invalid SOV address");
+		ILiquidityMining lm = ILiquidityMining(_lmContract);
+
+		(address[] memory _poolToken, uint96[] memory _allocationPoints, uint256[] memory _lastRewardBlock) = lm.getPoolInfoListArray();
+
+		require(_poolToken.length == _allocationPoints.length, "Arrays mismatch");
+		require(_poolToken.length == _lastRewardBlock.length, "Arrays mismatch");
+
+		for (uint256 i = 0; i < _poolToken.length; i++) {
+			address poolToken = _poolToken[i];
+			uint96[] memory allocationPoints = new uint96[](1);
+			allocationPoints[0] = _allocationPoints[i];
+			uint256 lastRewardBlock = _lastRewardBlock[i];
+			address[] memory SOVAddress = new address[](1);
+			SOVAddress[0] = _SOVAddress;
+			//add will revert if poolToken is invalid or if it was already added
+			add(poolToken, SOVAddress, allocationPoints, false);
```

Comment:
do we need to call `addRewardToken` before running this, right? maybe from the deployment script?

Reply:
Yes, that's right